### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by MadayAlcala**
I added new address scenario created, please review it.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -37,7 +37,10 @@ dependencies {
     compile group: 'info.cukes', name: 'cucumber-testng', version: '1.2.5'
     compile 'org.hamcrest:hamcrest:2.1'
     compile group: 'net.masterthought', name: 'cucumber-reporting', version: '4.6.0'
-    compile group: 'log4j', name: 'log4j', version: '1.2.17'
+    compile group: 'log4j', name: 'log4j', version: '1.2.16'
+    compile group: 'info.cukes', name: 'cucumber-picocontainer', version: '1.2.5'
+    // https://mvnrepository.com/artifact/org.slf4j/slf4j-simple
```
Comment (titor-oopart) [inline build.gradle]: delete this comment

File: config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -87,7 +75,7 @@
         <module name="LocalVariableName"/>
         <module name="MemberName"/>
         <module name="MethodName"/>
-<!--        <module name="PackageName"/>-->
+        <!--        <module name="PackageName"/>-->
```
Comment (titor-oopart) [inline config/checkstyle/checkstyle.xml]: align  text with the rest

File: config/checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -192,4 +180,4 @@
 
     </module>
 
-</module>
+</module>
```
Comment (titor-oopart) [inline config/checkstyle/checkstyle.xml]: add empty line

File: src/main/java/AutomationPractice/ui/StrategySetter.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+package AutomationPractice.ui;
+
+/**
+ * StrategySetter interface.
+ *
+ * @author Maday Alcala
+ * @version 0.0.1
+ */
+public interface StrategySetter {
+    /**
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/StrategySetter.java:21]: add empty line accord other classes

File: src/main/java/AutomationPractice/ui/entities/Address.java [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+ * Address class.
+ *
+ * @author Maday Alcala
+ * @version 0.0.1
+ */
+public class Address {
+    private String firstname;
+    private String lastname;
+    private String company;
+    private String address1;
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/entities/Address.java]: you can put a better name like mainAddress and secondAddress for address2

File: src/main/java/AutomationPractice/ui/entities/Address.java [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+     * @return phone.
+     */
+    private String getPhone() {
+        return phone;
+    }
+
+    /**
+     * This method is in charge of set the address phone.
+     *
+     * @param phone phone.
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/entities/Address.java]: explain better this param

File: src/main/java/AutomationPractice/ui/entities/Address.java [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+     * @return phoneMobile.
+     */
+    public String getPhoneMobile() {
+        return phoneMobile;
+    }
+
+    /**
+     * This method is in charge of set the address phoneMobile.
+     *
+     * @param phoneMobile phoneMobile.
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/entities/Address.java]: explain better this param

File: src/main/java/AutomationPractice/ui/entities/Address.java [hunk 8]
```diff
@@ -0,0 +1,330 @@
...
+     */
+    private void setPhoneMobile(final String phoneMobile) {
+        this.phoneMobile = phoneMobile;
+    }
+
+    /**
+     * This method is in charge of return the address other.
+     *
+     * @return other.
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/entities/Address.java]: the name of other can be otherAddress

File: src/main/java/AutomationPractice/ui/entities/Address.java [hunk 9]
```diff
@@ -0,0 +1,330 @@
...
+     */
+    private String getOther() {
+        return other;
+    }
+
+    /**
+     * This method is in charge of set the address other.
+     *
+     * @param other other.
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/entities/Address.java]: specify more "other"

File: src/main/java/core/selenium/webdrivers/Firefox.java [hunk 10]
```diff
@@ -1,3 +1,15 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/Firefox.java]: change the name of FormPageFour.java accord to your classes

File: src/main/java/core/selenium/webdrivers/Chrome.java [hunk 11]
```diff
@@ -1,3 +1,15 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/Chrome.java]: change the name of FormPageFour.java accord to your classes

File: src/main/java/core/selenium/webdrivers/BrowserType.java [hunk 12]
```diff
@@ -1,3 +1,15 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/BrowserType.java]: change the name of FormPageFour.java accord to your classes

File: src/main/java/core/selenium/WebDriverManager.java [hunk 13]
```diff
@@ -1,3 +1,15 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/main/java/core/selenium/WebDriverManager.java]: change the name of FormPageFour.java accord to your classes

File: src/main/java/core/selenium/WebDriverFactory.java [hunk 14]
```diff
@@ -1,3 +1,15 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/main/java/core/selenium/WebDriverFactory.java]: change the name of FormPageFour.java accord to your classes

File: src/main/java/core/selenium/webdrivers/IBrowser.java [hunk 15]
```diff
@@ -1,3 +1,15 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/IBrowser.java]: change the name of FormPageFour.java accord to your classes

File: src/main/java/core/utils/Log.java [hunk 16]
```diff
@@ -1,8 +1,19 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/main/java/core/utils/Log.java]: change the name of FormPageFour.java accord to your classes

File: src/test/java/hook/Hook.java [hunk 17]
```diff
@@ -1,5 +1,19 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/test/java/hook/Hook.java]: change the name of FormPageFour.java accord to your classes

File: src/test/java/runner/RunCukesTest.java [hunk 18]
```diff
@@ -1,3 +1,15 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/test/java/runner/RunCukesTest.java]: change the name of FormPageFour.java accord to your classes

File: src/test/java/steps/AddressSteps.java [hunk 19]
```diff
@@ -1,9 +1,26 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/test/java/steps/AddressSteps.java]: change the name of FormPageFour.java accord to your classes

File: src/test/java/steps/LoginSteps.java [hunk 20]
```diff
@@ -1,3 +1,15 @@
+/*
+ * @(#) FormPageFour.java Copyright (c) 2019 Jala Foundation.
```
Comment (titor-oopart) [inline src/test/java/steps/LoginSteps.java]: change the name of FormPageFour.java accord to your classes

File: src/test/resources/feature/address.feature [hunk 21]
```diff
@@ -1,8 +1,38 @@
...
-    When I update a "field" of the address
+  Background: Log in with valid credentials
+    Given The user goes to "login" page
+    When The user fills the form with "email" and "password"
+    Then Username should appear in the left panel
+
+  Scenario: Create new address with the required fields
+    When The user goes to create "address" form
+    And The user fills the form with the following information
+      | firstname  | firstname     |
```
Comment (titor-oopart) [inline src/test/resources/feature/address.feature]: put better names

File: build.gradle (hunk 1 above)
Comment (MadayAlcala) [reply build.gradle]: Solved.

File: config/checkstyle/checkstyle.xml (hunk 2 above)
Comment (MadayAlcala) [reply config/checkstyle/checkstyle.xml]: Solved.

File: config/checkstyle/checkstyle.xml (hunk 3 above)
Comment (MadayAlcala) [reply config/checkstyle/checkstyle.xml]: Solved.

File: src/main/java/AutomationPractice/ui/StrategySetter.java (hunk 4 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/StrategySetter.java:21]: Solved.

File: src/main/java/core/selenium/webdrivers/Firefox.java (hunk 10 above)
Comment (MadayAlcala) [reply src/main/java/core/selenium/webdrivers/Firefox.java]: Solved.

File: src/main/java/core/selenium/webdrivers/Chrome.java (hunk 11 above)
Comment (MadayAlcala) [reply src/main/java/core/selenium/webdrivers/Chrome.java]: Solved.

File: src/main/java/core/selenium/webdrivers/BrowserType.java (hunk 12 above)
Comment (MadayAlcala) [reply src/main/java/core/selenium/webdrivers/BrowserType.java]: Solved.

File: src/main/java/core/selenium/WebDriverManager.java (hunk 13 above)
Comment (MadayAlcala) [reply src/main/java/core/selenium/WebDriverManager.java]: Solved.

File: src/main/java/core/selenium/WebDriverFactory.java (hunk 14 above)
Comment (MadayAlcala) [reply src/main/java/core/selenium/WebDriverFactory.java]: Solved.

File: src/main/java/core/selenium/webdrivers/IBrowser.java (hunk 15 above)
Comment (MadayAlcala) [reply src/main/java/core/selenium/webdrivers/IBrowser.java]: Solved.

File: src/main/java/core/utils/Log.java (hunk 16 above)
Comment (MadayAlcala) [reply src/main/java/core/utils/Log.java]: Solved.

File: src/test/java/hook/Hook.java (hunk 17 above)
Comment (MadayAlcala) [reply src/test/java/hook/Hook.java]: Solved.

File: src/test/java/runner/RunCukesTest.java (hunk 18 above)
Comment (MadayAlcala) [reply src/test/java/runner/RunCukesTest.java]: Solved.

File: src/test/java/steps/AddressSteps.java (hunk 19 above)
Comment (MadayAlcala) [reply src/test/java/steps/AddressSteps.java]: Solved.

File: src/test/java/steps/LoginSteps.java (hunk 20 above)
Comment (MadayAlcala) [reply src/test/java/steps/LoginSteps.java]: Solved.

File: src/test/resources/feature/address.feature (hunk 21 above)
Comment (MadayAlcala) [reply src/test/resources/feature/address.feature]: Solved.

File: src/main/java/AutomationPractice/ui/entities/Address.java (hunk 5 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/entities/Address.java]: Solved.

File: src/main/java/AutomationPractice/ui/entities/Address.java (hunk 8 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/entities/Address.java]: Solved.

File: src/main/java/AutomationPractice/ui/entities/Address.java (hunk 6 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/entities/Address.java]: Solved.

File: src/main/java/AutomationPractice/ui/entities/Address.java (hunk 7 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/entities/Address.java]: Solved.

File: src/main/java/AutomationPractice/ui/entities/Address.java (hunk 9 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/entities/Address.java]: Solved.

File: src/main/java/AutomationPractice/ui/Common/ReadAppFile.java [hunk 22]
```diff
@@ -15,6 +22,7 @@
     private static final String EMAIL = "email";
     private static final String PASSWORD = "password";
     private static final String PROPERTIES_FILE = "automationpractice.properties";
+
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/Common/ReadAppFile.java]: delete empty line

File: src/main/java/AutomationPractice/ui/Common/UrlBuilder.java [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+    static final String URL_BASE = "url";
+    static final String LOGIN = "controller=authentication&back=my-account";
+    static final String ADDRESS = "controller=address";
+    static final String WISHLIST = "fc=module&module=blockwishlist&controller=mywishlist";
+    static final String DRESSES = "id_category=8&controller=category";
+    static final String CASUAL_DRESSES = "id_category=9&controller=category";
+    static final String SUMMER_DRESSES = "id_category=11&controller=category";
+    static final String EVENING_DRESSES = "id_category=10&controller=category";
+    static final String CONTROLLER_ORDER = "controller=order";
+
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/Common/UrlBuilder.java:32]: delete unnecessary empty lines

File: src/main/java/AutomationPractice/ui/Common/UrlBuilder.java [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+        map.put("login", ReadAppFile.getInstance().getUrl(URL_BASE).concat(LOGIN));
+        map.put("address", ReadAppFile.getInstance().getUrl(URL_BASE).concat(ADDRESS));
+        map.put("wishlist", ReadAppFile.getInstance().getUrl(URL_BASE).concat(WISHLIST));
+        map.put("dresses", ReadAppFile.getInstance().getUrl(URL_BASE).concat(DRESSES));
+        map.put("casualDresses", ReadAppFile.getInstance().getUrl(URL_BASE).concat(CASUAL_DRESSES));
+        map.put("summerDresses", ReadAppFile.getInstance().getUrl(URL_BASE).concat(SUMMER_DRESSES));
+        map.put("eveningDresses", ReadAppFile.getInstance().getUrl(URL_BASE).concat(EVENING_DRESSES));
+        map.put("controlOrder", ReadAppFile.getInstance().getUrl(URL_BASE).concat(CONTROLLER_ORDER));
+
+
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/Common/UrlBuilder.java]: delete unnecessary empty lines

File: src/main/java/AutomationPractice/ui/pages/Account.java [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+ * @version 0.0.1
+ */
+public class Account extends BasePage {
+
+    @FindBy(xpath = "//a[@title=\"View my customer account\"]")
+    private WebElement accountBtn;
+
+    @FindBy(xpath = "//span[contains(text(), \"Order history and details\")]")
+    private WebElement accountOrderHistoryBtn;
+
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/pages/Account.java:32]: delete empty line

File: src/main/java/AutomationPractice/ui/pages/Account.java [hunk 26]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * This methos is used for obtain the order list.
+     *
+     * @return a list of 'orderList'.
+     */
+    public List<WebElement> getAccountOrdersLis() {
+        return driver.findElements(By.xpath("//table[@id=\"order-list\"]/tbody/tr"));
+    }
+
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/pages/Account.java:41]: delete line

File: src/main/java/AutomationPractice/ui/pages/AddressFormAbstract.java [hunk 27]
```diff
@@ -0,0 +1,264 @@
...
+    public abstract void setFirstName(String firstName);
+
+    /**
+     * This method is in charge of return the address' first name.
+     *
+     * @return first name.
+     */
+    public abstract String getFirstNameTxt();
+
+
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/pages/AddressFormAbstract.java]: delete line

File: src/main/java/AutomationPractice/ui/pages/AddressPage.java [hunk 28]
```diff
@@ -130,61 +215,100 @@ private void setcity(final String city) {
...
      *
      * @param other 'other', that represent an other.
      */
-    private void setOther(final String other) {
+    @Override
+    public void setOther(final String other) {
         othertxtField.sendKeys(other);
     }
 
+
```
Comment (titor-oopart) [inline src/main/java/AutomationPractice/ui/pages/AddressPage.java]: delete line

File: src/test/java/steps/ShoppingSteps.java [hunk 29]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    /**
+     * This method is for choose a payment meethod.
+     */
+    @When("^The user choose a payment method$")
+    public void choosingPaymentMethod() {
+        summary.getCartSummPayByBankWire();
+        Assert.assertEquals(summary.getCartSummPayByBankWireConfirm(), "BANK-WIRE PAYMENT.");
+
```
Comment (titor-oopart) [inline src/test/java/steps/ShoppingSteps.java]: delete line

File: src/test/java/steps/ShoppingSteps.java [hunk 30]
```diff
@@ -0,0 +1,149 @@
...
+    /**
+     * This method is for check the succes message.
+     *
+     * @param successMessage of the order.
+     */
+    @Then("^The user should see this success message: \"([^\"]*)\"$")
+    public void successMessage(final String successMessage) {
+        summary.getCartSummConfirmOrderBtn();
+
```
Comment (titor-oopart) [inline src/test/java/steps/ShoppingSteps.java]: delete line

File: src/test/resources/feature/address.feature [hunk 31]
```diff
@@ -1,8 +1,40 @@
-Feature: Address
-  Background:
-    Given I visit the "login" page
-    When I fill the form with "email" and "password"
-
-  Scenario: Edit the address as registered user
-    Given I go to update "address" form
-    When I update a "field" of the address
+#Feature: Address
```
Comment (titor-oopart) [inline src/test/resources/feature/address.feature]: delte this feature

File: src/test/resources/feature/login.feature [hunk 32]
```diff
@@ -1,12 +1,14 @@
...
-    Then Username should appear in the left panel
-      And "Sign out" should appear
-
-  Scenario: Login as registered user
-    Given I visit the "login" page
-    When I fill the form with "email" and "password"
-    Then Username should appear in the left panel
-      And "Sign out" should appear
+#Feature: Login
+#  @signOut
```
Comment (titor-oopart) [inline src/test/resources/feature/login.feature]: delte feature

File: src/test/resources/feature/shopping.feature [hunk 33]
```diff
@@ -0,0 +1,25 @@
...
+      | TotalProductsPrice | $105.97 |
+      | TotalPrice         | $107.97 |
+      | TotalShipping      | $2.00   |
+
+    When The user accept the terms of service check
+      And The user choose a payment method
+    Then The user should see this success message: "Your order on My Store is complete."
+      And The user should see the order history with the number 41
+
+
```
Comment (titor-oopart) [inline src/test/resources/feature/shopping.feature]: delte line

File: src/main/java/AutomationPractice/ui/Common/ReadAppFile.java (hunk 22 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/Common/ReadAppFile.java]: Solved.

File: src/main/java/AutomationPractice/ui/Common/UrlBuilder.java (hunk 23 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/Common/UrlBuilder.java:32]: Solved.

File: src/main/java/AutomationPractice/ui/Common/UrlBuilder.java (hunk 24 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/Common/UrlBuilder.java]: Solved.

File: src/main/java/AutomationPractice/ui/pages/Account.java (hunk 25 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/pages/Account.java:32]: Solved.

File: src/main/java/AutomationPractice/ui/pages/Account.java (hunk 26 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/pages/Account.java:41]: Solved.

File: src/main/java/AutomationPractice/ui/pages/AddressFormAbstract.java (hunk 27 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/pages/AddressFormAbstract.java]: Solved.

File: src/main/java/AutomationPractice/ui/pages/AddressPage.java (hunk 28 above)
Comment (MadayAlcala) [reply src/main/java/AutomationPractice/ui/pages/AddressPage.java]: solved.

File: src/test/java/steps/ShoppingSteps.java (hunk 29 above)
Comment (MadayAlcala) [reply src/test/java/steps/ShoppingSteps.java]: solved.

File: src/test/java/steps/ShoppingSteps.java (hunk 30 above)
Comment (MadayAlcala) [reply src/test/java/steps/ShoppingSteps.java]: solved.

File: src/test/resources/feature/address.feature (hunk 31 above)
Comment (MadayAlcala) [reply src/test/resources/feature/address.feature]: solved.

File: src/test/resources/feature/login.feature (hunk 32 above)
Comment (MadayAlcala) [reply src/test/resources/feature/login.feature]: solved.

**Review by titor-oopart — APPROVED** [review]

**Pull request merged**